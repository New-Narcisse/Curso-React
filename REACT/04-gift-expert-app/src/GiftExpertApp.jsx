import { useState } from "react";
import { AddCategory, GiftGrid } from "./components";

export const GiftExpertApp = () => {
  const [categories, setCategories] = useState([""]);
  // console.log(categories);
  const onAddCategory = (newCategory) => {
    /* esta funcion es para agregar lo que se me antoje a travez del boton de agregar o enter*/
    if (categories.includes(newCategory)) return;
    //    console.log('Kimetsu no yaiba')
    setCategories([newCategory, ...categories]);
    //setCategories((cat) => [...cat, newCategory]);
    /* cualquiera de las dos opciones son validas */
  };

  return (
    <div>
      <h1>GiftExpertApp</h1>

      <AddCategory onNewCategory={(value) => onAddCategory(value)} />

      {categories.map((categori) => (
        <GiftGrid key={categori} category={categori} />
      ))}
    </div>
  );
};

//setCategories={setCategories}
/* setCategories={setCategories} es una metodologia que oculta 
loque realmente esta sucediendo por lo tanto no es evidente su lectura */
// el prefijo on significa que se esta emitiendo algo es decir, un evento
/* en el siguinete codigo: onNewCategory={ ( event ) => onNewCategory(event) } podemos por ley eliminar la sintaxis del event 
ya que estan en ambos lados de la funcion. */
/* RECURDA QUE EL 'onNewCategory' ES UNA PROPIEDAD DE MI COMPONENTE*/
/* <div key={category}>
  <h3> {category} </h3>
  <li>{category}</li>
</div> */
