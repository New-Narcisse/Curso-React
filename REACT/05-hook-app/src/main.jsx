// import { CounterApp } from "./01-useState/CounterApp";
// import { CounterWithCustomHooks } from "./01-useState/CounterWithCustomHooks";
// import { HooksApp } from "./HooksApp";
// import { SimpleForm } from "./02-useEffect/SimpleForm";
// import { SimpleFormWithHook } from "./02-useEffect/SimpleFormWithHook";
// import { SimpleFormWithHook } from "./02-useEffect/SimpleFormWithHook";
// import { MultipleCustumHooks } from "./03-example/MultipleCustumHooks";
// import { FocusScreen } from "./04-useRef/FocusScreen";
// import { Memorize } from "./06-memo/Memorize";
// import { MemorizeHook } from "./06-memo/MemorizeHook";
// import { CallbackHook } from "./06-memo/CallbackHook";
// import { Padre } from "./07-tarea-memo/Padre";
// import "./08-useReducer/intro-reducer";
// import { TodoApp } from "./08-useReducer/TodoApp";
import { BrowserRouter } from "react-router-dom";

import React from "react";
import ReactDOM from "react-dom/client";
import "./index.css";
import { MainApp } from "./09-useContext/MainApp";
ReactDOM.createRoot(document.getElementById("root")).render(
  <BrowserRouter>
    {/* <React.StrictMode> */}
    <MainApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>
);
