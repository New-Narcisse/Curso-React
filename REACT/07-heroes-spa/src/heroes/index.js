


export * from './pages';
export * from './helpers';
export * from './componentes';
export * from './routes/HeroesRoutes';

