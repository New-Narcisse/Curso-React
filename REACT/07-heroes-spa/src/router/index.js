



export * from './AppRouter'
export * from './PrivatRoute'
export * from './PublicRoute'