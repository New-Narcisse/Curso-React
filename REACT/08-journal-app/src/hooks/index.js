export * from './useForm';
export * from './useCheckauth';


