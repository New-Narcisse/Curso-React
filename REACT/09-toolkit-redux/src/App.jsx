import { useDispatch, useSelector } from "react-redux";
import reactLogo from "./assets/react.svg";
import viteLogo from "/vite.svg";
import "./App.css";
import { decrement, increment, incrementBy } from "./store";

function App() {
  const { counter } = useSelector((state) => state.counter);

  const dispatch = useDispatch();

  return (
    <>
      <div>
        <a href="https://vitejs.dev" rel="noreferrer" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" rel="noreferrer" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>count is {counter}</h1>
      <div className="card">
        <button onClick={() => dispatch(increment())}> Increment </button>
      </div>
      <div className="card">
        <button onClick={() => dispatch(decrement())}> Decrement </button>
      </div>
      <div className="card">
        <button onClick={() => dispatch(incrementBy(2))}> IncrementBy </button>
      </div>
    </>
  );
}

export default App;
