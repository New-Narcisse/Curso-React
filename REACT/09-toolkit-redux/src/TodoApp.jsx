import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/apis";

export const TodoApp = () => {
const [todoId, setTodoId] = useState(1)

  
  const { data: todo, isLoading } = useGetTodoQuery(todoId);

  const nextTodo = () => {
    setTodoId(todoId + 1);
  }

  const prevTodo = () => {
    if (todoId === 1) return;
    setTodoId( todoId - 1 )
  }



  return (
    <>
      <h1> Todos - RTK Query </h1>
      <hr />
      <h4> isLoading: {isLoading ? "True" : "False"} </h4>

      <pre> {JSON.stringify(todo)} </pre>

      <button onClick={nextTodo} >Next todo</button>
      <button onClick={prevTodo} >Prev todo</button>
      {/* <ul>
        {todos.map((data) => (
          <li key={data.id}>
            <strong> {data.completed ? "DONE" : "PENDING"} </strong>
            {data.title}
          </li>
        ))}
      </ul> */}
    </>
  );
};
