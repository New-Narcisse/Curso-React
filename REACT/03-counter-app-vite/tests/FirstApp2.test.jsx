import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en FirstApp", () => {
  const title = "Hola, Soy Sony";
  const subTitle = "Soy un sub titulo";

  test("should match with the snap", () => {
    const { container } = render(<FirstApp title={title} />);
    //screen.debug()
    expect(container).toMatchSnapshot();
  });

  test("should show Hola, soy Sony", () => {
    /* No es necesario crear la constante 'title' dentro del test
    ya que fue creado en el describe, de esa manera todos los 
    test lo podran usar por invocacion*/
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("show the title in an h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("should show the subtitle sent by props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(1);
  });
});
