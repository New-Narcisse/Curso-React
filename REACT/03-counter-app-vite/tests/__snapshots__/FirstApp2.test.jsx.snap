// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should match with the snap 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Sony
     
  </h1>
  <p>
     
    No hay sub titulos
     
  </p>
  <p>
     
    Sony Narcisse
     
  </p>
</div>
`;
